js/apps/admin-ui/src/identity-providers/add/GeneralSettings.tsx

```diff
@@ -1,19 +1,31 @@
 import { RedirectUrl } from "../component/RedirectUrl";
 import { ClientIdSecret } from "../component/ClientIdSecret";
+import { TextField } from "../component/TextField";
 import { DisplayOrder } from "../component/DisplayOrder";
 
+
 type GeneralSettingsProps = {
   id: string;
+  showclientidsecret?: boolean;
   create?: boolean;
 };
 
 export const GeneralSettings = ({
   create = true,
+  showclientidsecret = true,
   id,
-}: GeneralSettingsProps) => (
-  <>
+}: GeneralSettingsProps) => {
+  return (
+  
+    <>
     <RedirectUrl id={id} />
-    <ClientIdSecret create={create} />
+    {showclientidsecret && (
+      <>
+        <ClientIdSecret create={create} />
+      </>
+    )}
+    <TextField field="displayName" label="displayName" />
```

Comment:
This new field does not seems related to the original bug that was was reported. I think an enhancement like this would warrant further discussion, so I'd prefer to handle this in a separate issue and PR.